cfme/cloud/flavor.py

```diff
@@ -108,11 +143,66 @@ class Flavor(BaseEntity, Taggable):
     name = attr.ib()
     provider = attr.ib()
 
+    def delete(self, wait=True):
+        """Delete current falvor """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove Flavor', handle_alert=True)
+        view.flash.assert_success_message(
+            'Delete of Flavor "{}" was successfully initiated.'.format(self.name))
+
+        if wait:
+            self.wait_for_disappear()
+
+    def wait_for_disappear(self):
+        """Wait for flavor to disappear"""
+        wait_for(self.provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=300)
+        wait_for(lambda: not self.exists,
+                 timeout=300,
+                 message='Wait for flavor to disappear',
+                 delay=20,
+                 fail_func=self.browser.refresh)
```

Comment:
Thought... why not call self.refresh here, which calls the method ```refresh()``` on line 165 below, then we could possible remove the wait for on line 158. What do you think? If you do agree, I think moving the wait into the delete method would be the next step.